cnosdb/cnosdb
# [Test] Add coordinator cases

# Rationale for this change

Add tests for coordinator, these tests are run in series.

# What changes are included in this PR?

- Add constructors for `DatabaseSchema` and `DatabaseOptions`.
- Add another integration test `Integration Test cn` for to run `coordinator_tests::tests` in package `e2e_test`.
- The integration test `Integration Test Singleton` now only run `http_api_tests::test` and `kv_service_test::test` in package `e2e_test`.

## e2e_test/../coordinator_tests.rs

Add e2e test cases to test coordinator and github action `Integration Test cn` to run these cases.

- `test_multi_tenants_write_data`
  - Kill all existing processes that named  `cnosdb-meta` or `cnosdb` .
  - Run 1m2d cluster (includes cleaning old files).
  - Call `prepare()` - create 5 tenants and databases.
  - Call `prepare_data()` - create 5 thread, for each thread:
    - Generate request HTTP URL, with one of the created tenants and databases.
    - Write 100 points using generated URL.
  - Check points count for each tenant-databases.
  - Kill all existing processes that named  `cnosdb-meta` or `cnosdb`
- `test_replica` - Number of copies of  the test database;
- `test_shard` - Number of shards of the test database;
- `test_ttl` - Test database ttl function;
- `test_balance` - Test database load balancing (TODO);
